scvi/external/wscvi/_module.py

```diff
@@ -0,0 +1,270 @@
+from typing import Callable, Iterable, Optional
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.distributions import Normal
+
+from scvi import REGISTRY_KEYS
+from scvi._compat import Literal
+from scvi.module import VAE
+from scvi.module.base import LossRecorder, auto_move_data
+from scvi.nn import one_hot
+
+
+def reparameterize_gaussian(mu, var):
+    return Normal(mu, var.sqrt()).rsample()
+
+
+class WVAE(VAE):
```

Comment:
This is a lot of duplicate code just for multiple particles and a modified loss function. I think moving the n_particles portion (default to 1) to VAE makes sense then having WVAE just overwrite the loss function (after addressing below comment).

After that change, if we believe in this method enough, I think it would be worthwhile making this an option (switch out the module) in SCVI rather than as an external model, considering the DEMixin functions are shared with SCVI anyway.